ui/blueprints/story/index.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 const getPathOption = require('ember-cli-get-component-path-option');
 const stringUtil = require('ember-cli-string-utils');
+const path = require('path');
+
+function findAddonByName(addonOrProject, name) {
+  let addon = addonOrProject.addons.find(addon => addon.name === name);
+
+  if (addon) {
+    return addon;
+  }
+
+  return addonOrProject.addons.find(addon => findAddonByName(addon, name));
+}
 
 module.exports = {
   description: 'generates a story for storybook',
 
   fileMapTokens: function() {
+    let { project } = this;
     return {
+      __path__: function(options) {
+        if (options.inRepoAddon) {
+          let addon = findAddonByName(project, options.inRepoAddon);
```

Comment:
isn't there a possibility that the addon wouldn't be found, `findAddonByName` would return `-1` and it wouldn't have a `.root`? How does this handle that case?

Reply:
I wasn't sure so I tried this out, and got this message:

```
ember g story flash-message -ir foo 
installing story
You specified the 'in-repo-addon' flag, but the in-repo-addon 'foo' does not exist. Please check the name and try again.
```

✨